src/simulator.zig

```diff
@@ -222,7 +231,8 @@ fn client_callback(
     assert(user_data == 0);
 }
 
-fn parse_seed(bytes: []const u8) u64 {
+fn parse_seed(bytes: []const u8, ga: *std.mem.Allocator) u64 {
+    defer ga.free(bytes);
```

Comment:
We should rather move this `free()` call to be next to (i.e. in the same scope) where we receive the allocated value, so that it's more clear where we take ownership and where we release. Let's make line 37 above a labelled block that returns a value when we break, so that we can keep this all together. That would also enable us to remove the need to introduce the `next_arg` function. Everything can just be in the same block scope and then it's pretty clear.